PyOpenWorm/simpleProperty.py

```diff
@@ -223,34 +290,76 @@ def has_defined_value(self):
         return self._pp.has_defined_value()
 
     def __call__(self, *args, **kwargs):
-        """ If arguments are passed to the ``Property``, its ``set`` method
-        is called. Otherwise, the ``get`` method is called. If the ``multiple``
-        member for the ``Property`` is set to ``True``, then a Python set containing
-        the associated values is returned. Otherwise, a single bare value is returned.
-        """
-
-        if len(args) > 0 or len(kwargs) > 0:
-            self.set(*args, **kwargs)
-            return self
-        else:
-            r = self.get(*args, **kwargs)
-            if self.multiple:
-                return set(r)
-            else:
-                try:
-                    return next(iter(r))
-                except StopIteration:
-                    return None
+        return _get_or_set(self, *args, **kwargs)
 
     def one(self):
-        l = list(self.get())
-        if len(l) > 0:
-            return l[0]
-        else:
-            return None
+        return _next_or_none(self.get())
 
     @classmethod
     def register(cls):
         cls.rdf_type = cls.conf['rdf.namespace'][cls.__name__]
         cls.rdf_namespace = R.Namespace(cls.rdf_type + "/")
         cls.conf['rdf.namespace_manager'].bind(cls.__name__, cls.rdf_namespace)
+
+
+def _get_or_set(self, *args, **kwargs):
+    """ If arguments are given ``set`` method is called. Otherwise, the ``get``
+    method is called. If the ``multiple`` member is set to ``True``, then a
+    Python set containing the associated values is returned. Otherwise, a
+    single bare value is returned.
+    """
+    # XXX: checking this in advance because I'm paranoid, I guess
```

Comment:
Make _get_or_set a mixin so the asserts can go into an __init__ rather than checking on every call.